### PR TITLE
add basic String#localeCompare

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/string.ts [hunk 1]
```diff
@@ -163,6 +163,18 @@ import { idof } from "./builtins";
     return -1;
   }
 
+  // TODO: implement full locale comparison with locales and Collator options
+  localeCompare(other: String): i32 {
+    if (other === this) return 0; // compare pointers
+    if (other === null) other = changetype<String>("null");
```
Comment (dcodeIO) [inline std/assembly/string.ts]: Seems this can't happen

Comment (MaxGraey) [reply std/assembly/string.ts]: right!

File: std/assembly/util/string.ts [hunk 2]
```diff
@@ -101,11 +101,12 @@ export function compareImpl(str1: string, index1: usize, str2: string, index2: u
 }
 
 export function isSpace(c: i32): bool {
-  if (c <= 0xFF) {
+  if (c < 0x1680) { // < <LS> (1)
     // <SP>, <TAB>, <LF>, <VT>, <FF>, <CR> and <NBSP>
-    return c == 0x20 || <u32>(c - 0x09) <= 0x0D - 0x09 || c == 0xA0;
+    // @ts-ignore: cast
+    return ((c | 0x80) == 0xA0) | (u32(c - 0x09) <= 0x0D - 0x09);
```
Comment (dcodeIO) [inline std/assembly/util/string.ts]: Can you explain what changed here?

Comment (MaxGraey) [reply std/assembly/util/string.ts]: Sure. If we slightly rearrange original condition as:
```ts
(c == 0x20 || c == 0xA0) || <u32>(c - 0x09) <= 0x0D - 0x09
```
and convert `0x20` and `0xA0` to binary:
```ts
0b00100000 // 0x20
0b10100000 // 0xA0
// also
0b10000000 // 0x80
```
we could see `(c | 0b10000000) == 0b10100000` always `true` for `0b00100000` or `0b10100000`

Comment (MaxGraey) [reply std/assembly/util/string.ts]: Added comment

**Pull request merged**